integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/store/AbstractInfinispanStoreRocksDBIT.java

```diff
@@ -21,41 +24,66 @@
  * @since 7.0
  */
 public abstract class AbstractInfinispanStoreRocksDBIT {
-   private static String baseDir = tmpDirectory(AbstractInfinispanStoreRocksDBIT.class);
-   private static String dataDir = baseDir + separator + "data";
-   private static String expiredDir = baseDir + separator + "expired";
-
-   private EmbeddedCacheManager cm;
-
-   @Before
-   @After
-   public void removeDataFilesIfExists() {
-      Util.recursiveFileRemove(baseDir);
-      if (cm != null)
-         cm.stop();
+
+   @ArquillianResource
+   private Deployer deployer;
+
+   @Test
+   @InSequence(1)
+   public void deployDep1() {
+      deployer.deploy("dep1");
    }
 
-   /**
-    * To avoid pulling in TestingUtil and its plethora of dependencies
-    */
-   private static String tmpDirectory(Class<?> test) {
-      String prefix = System.getProperty("infinispan.test.tmpdir", System.getProperty("java.io.tmpdir"));
-      return prefix + separator + "infinispanTempFiles" + separator + test.getSimpleName();
+   @Test
+   @OperateOnDeployment("dep1")
+   @InSequence(2)
+   public void testRunningInDep1() {
+      DefaultCacheManager cm = createCacheManager();
+      Cache<String, String> cache = cm.getCache();
+      cache.put("dep1", "dep1");
+   }
+
+   @Test
+   @InSequence(3)
+   public void deployDep2() {
+      deployer.deploy("dep2");
    }
 
    @Test
-   public void testCacheManager() {
-      GlobalConfigurationBuilder gcb = new GlobalConfigurationBuilder();
-      gcb.globalState().persistentLocation(baseDir).defaultCacheName("default");
+   @OperateOnDeployment("dep2")
+   @InSequence(4)
+   public void testRunningInDep2() {
+      DefaultCacheManager cm = createCacheManager();
+      Cache<String, String> cache = cm.getCache();
+      assertEquals("dep1", cache.get("dep1"));
+   }
+
+   @Test
+   @InSequence(5)
+   public void undeploy() {
+      deployer.undeploy("dep1");
+      deployer.undeploy("dep2");
+   }
+
+   protected DefaultCacheManager createCacheManager() {
+      String prefix = System.getProperty("infinispan.test.tmpdir", System.getProperty("java.io.tmpdir"));
+      String baseDir = prefix + separator + "infinispanTempFiles" + separator + this.getClass().getSimpleName() + separator + UUID.randomUUID().toString();
```

Comment:
Done

Reply:
@diegolovison Did you forget to push these changes?

Reply:
It was
```
String prefix = System.getProperty("infinispan.test.tmpdir", System.getProperty("java.io.tmpdir"));
```
It is
```
String prefix = CommonsTestingUtil.tmpDirectory("server-" + index);
```

Reply:
I see! The use of `infinispanTempFiles` confused me. In that case I think what @danberindei was suggesting is more like the following:

```java
String dataDir = CommonsTestingUtil.tmpDirectory(this.getClass().getSimpleName(), index, "data");
String expiredDir = CommonsTestingUtil.tmpDirectory(this.getClass().getSimpleName(), index, "expired");
```

Reply:
Could be
```
String baseDir = CommonsTestingUtil.tmpDirectory("server-" + index, Main.class.getSimpleName());
String dataDir = Paths.get(baseDir, "data").toString();
String expiredDir = Paths.get(baseDir, "expired").toString();
```

In this case, the files will go under the same bucket

Reply:
@diegolovison I'd rather put the class name first, so that all the files of a test class are in directory "className/server-*" instead of "server-*/className"
Otherwise I'm fine if you don't trust `tmpDirectory` and you'd rather concatenate the paths yourself ;)

Reply:
Hi @danberindei , do you mean ?
```
         String baseDir = CommonsTestingUtil.tmpDirectory(this.getClass().getSimpleName(), "server-" + index, UUID.randomUUID().toString());
         String dataDir = baseDir + separator + "data";
         String expiredDir = baseDir + separator + "expired";
```

Reply:
@diegolovison I meant just `tmpDirectory(this.getClass().getSimpleName(), "server-" + index)`, without a random UUID. Why do you think the UUID is needed?

Reply:
Ok. The UUID is not needed. Removed